src/components/App/ExpressRecorder.tsx

```diff
@@ -397,6 +397,9 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
             const screenBlobObject = new Blob(screenBlobs, { type: "video/webm" });
             fixWebmDuration(screenBlobObject, duration, (fixedBlob: Blob) => {
                 this.setState({ screenRecordedBlobs: screenBlobs, screenRecordedBlob: fixedBlob });
+                if (!recordedBlobs.length) {
+                    this.dispatcher.dispatchEvent(RecorderEvents.recordingEnded);
+                }
```

Comment:
on line **388** you have a condition for `if (recordedBlobs.length) { ... }` - so all required logic happens if the array is not empty.
so, why do you need another condition here? we know that this section refers to an empty array anyway...

Reply:
it's 2 different conditions - one for recordedBlobs, one for screenBlobs

Reply:
what do you mean?
I can see:
line **388** `if (recordedBlobs.length) {`
line **400** `if (!recordedBlobs.length) {`
so, the condition is upon the very same object

Reply:
yes but the second condition is not on the scope of the first one

Reply:
you right it can be better. I did something similar here - https://github.com/kaltura/express-recorder/pull/148/files
I'll merge this and change it on the other PR